src/pokedex.c

```diff
@@ -4409,7 +4409,7 @@ bool16 HasAllMons(void)
 
     for (i = 1; i < NATIONAL_DEX_COUNT + 1; i++)
     {
-        if (!(gSpeciesInfo[i].isMythical) && !GetSetPokedexFlag(i, FLAG_GET_CAUGHT))
+        if (!(gSpeciesInfo[i].isMythical && !gSpeciesInfo[i].dexForceRequired) && !GetSetPokedexFlag(i, FLAG_GET_CAUGHT))
             return FALSE;
     }
```

Comment:
you're using nat dex idexes in gSpeciesInfo.

Reply:
GF started :D what would be better here? There doesn't seem to be an existing function to convert nat dex indexes to species

Reply:
@AsparagusEduardo 👀

Reply:
`NationalPokedexNumToSpecies`

Reply:
> GF started

It was my fault, I'll take the L here xD
![image](https://github.com/rh-hideout/pokeemerald-expansion/assets/2904965/df8c212b-1a97-494b-b164-f2357df24d9f)